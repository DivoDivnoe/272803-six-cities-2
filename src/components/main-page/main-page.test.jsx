import React from 'react';
import renderer from 'react-test-renderer';
import MainPage from './main-page.jsx';

jest.mock(`../map/map.jsx`, () => jest.fn().mockReturnValue(null));

describe(`MainPage component`, () => {
  it(`is rendered correctly`, () => {
    const offers = [
      {
        id: 1,
        city: {
          name: `somecity`,
          location: {
            latitude: 0,
            longitude: 0
          }
        },
        type: `apartment`,
        price: 0,
        rating: 0,
        picture: `some/src`,
        images: [`some/src`],
        title: `some title`,
        goods: [`first`, `second`],
        bedrooms: 0,
        maxAdults: 0,
        description: `some desc.`,
        isPremium: true,
        host: {
          id: 1,
          isPro: true,
          name: `Angelina`,
          avatarUrl: `some/src`
        },
        location: {
          latitude: 0,
          longitude: 0
        }
      }
    ];

    const leaflet = jest.genMockFromModule(`leaflet`);

    const tree = renderer
      .create(
          <MainPage
            offers={offers}
            leaflet={leaflet}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
