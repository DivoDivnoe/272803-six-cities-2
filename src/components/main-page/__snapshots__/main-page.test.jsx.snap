// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component is rendered correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             place
            
             to stay in 
            somecity
          </b>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
            style={
              Object {
                "background": "none",
              }
            }
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
